### DISCUSSION TITLE
Potential problem-solving addition to the documentation?

### DISCUSSION THREAD

**Discussion by richardlovas** [JanDeDobbeleer/oh-my-posh#1760] [Ideas]
Hi all!

I'm a brand new oh-my-posh (and PowerShell Core) user, just installed it and don't even have my own config yet :) So please, be gentle with me if what I'm writing now is common knowledge. 

**Long post, so TL;DR is on the bottom.**

I installed oh-my-posh on two computers, my own laptop and my desktop at work.

**System configuration to understand the differences**

_Own laptop_
```
OS: Windows 10 Pro (latest updates installed)
OS display language: English
OS locale settings: hu_HU
PowerShell version: 7.2.1
Username: rezrex (local user, not MS account -- you'll see why I included this)
```

_Work desktop_
```
OS: Windows 10 Pro (latest updates installed)
OS display language: Hungarian
OS locale settings: hu_HU
PowerShell version: 7.2.1
Username: Gazdasági (local user, not MS account -- you'll see why I included this)
```

On the work computer, I've run into a problem that baffled me for a day or so, especially since following the same installation instructions on my own machine didn't reproduce it. 

On the work machine, I could run oh-my-posh as a command from Powershell, yet when I added the necessary line to my $PROFILE, PowerShell acted as if oh-my-posh isn't on the path.

![Képernyőkép 2022-02-10 121854](https://user-images.githubusercontent.com/99406926/153863483-bb57a19b-b5d5-425c-a3f0-42217726e6e6.png)

![Képernyőkép 2022-02-10 121926](https://user-images.githubusercontent.com/99406926/153863512-86c60cd6-fe1e-42e0-9bda-267cdc7b26d4.png)

![Képernyőkép 2022-02-10 122001](https://user-images.githubusercontent.com/99406926/153863537-6462516e-ab4a-4a53-b1d2-f0e21ada9a3d.png)

Eventually, I came to the realization that the "á" character in the username on that machine is the problem. For some reason, while PowerShell could perfectly parse this accented character with my locale settings, the initialization script of oh-my-posh in my $PROFILE couldn't handle it. 

From the error message and what I understand from the init script, this is most probably a PowerShell problem. It seems that accented characters in the path are handled differently when running a script form running an interactive shell.

The solution was to enable the Beta UTF-8 feature in the Windows locale setting. However, this borked some Hungarian applications I need for my work, so I had to disable this setting. 

After some search, I discovered the real solution that so far seems to have no problematic side effect. It's this line in my $PROFILE:

`$OutputEncoding = [console]::InputEncoding = [console]::OutputEncoding = New-Object System.Text.UTF8Encoding`

![Képernyőkép 2022-02-11 125822](https://user-images.githubusercontent.com/99406926/153866223-ac2e7501-157c-4ad3-95cf-7577ccb6b434.png)

I ask that someone more qualified check whether this solution has indeed no serious (or any) side effects. If so, this should probably be added to the documentation for those using a different locale than en_US without the UTF-8 beta feature enabled.

---------------

**TL;DR:** 

Non en_US locale withouth the beta UTF-8 setting on Windows 10 and non-English character in the username (or anywhere on oh-my-posh's full path) causes problems with the oh-my-posh initialization script in $PROFILE. 

Found a solution for those who also have problem with the global UTF-8 setting and legacy applications so they can't enable that. 

Someone should check whether the solution is really sound and then maybe this should be added to the documentation.

Thank you and I hope that this will really be helpful!

**Comment by JanDeDobbeleer** [comment]
Hey @richardlovas thanks for the suggestion. We're actually aware of this, but you're absolutely right that it should be added to the FAQ!

**Reply by richardlovas** [reply]
Hey @JanDeDobbeleer I knew that this is most likely a known thing. I almost didn't post it, but I'm glad that you also think this has a place in the FAQ! Thanks for the quick reply!
